vm/src/builtins/tuple.rs

```diff
@@ -254,36 +283,6 @@ impl PyTuple {
         };
         (tup_arg,)
     }
-
-    #[pyslot]
-    fn tp_new(
-        cls: PyTypeRef,
-        iterable: OptionalArg<PyObjectRef>,
-        vm: &VirtualMachine,
-    ) -> PyResult<PyRef<Self>> {
-        let elements = if let OptionalArg::Present(iterable) = iterable {
-            let iterable = if cls.is(&vm.ctx.types.tuple_type) {
-                match iterable.downcast_exact::<Self>(vm) {
-                    Ok(tuple) => return Ok(tuple),
-                    Err(iterable) => iterable,
-                }
-            } else {
-                iterable
-            };
-            vm.extract_elements(&iterable)?
-        } else {
-            vec![]
-        };
-        // Return empty tuple only for exact tuple types if the iterable is empty.
-        if elements.is_empty() && cls.is(&vm.ctx.types.tuple_type) {
-            Ok(vm.ctx.empty_tuple.clone())
-        } else {
-            Self {
-                elements: elements.into_boxed_slice(),
-            }
-            .into_ref_with_type(vm, cls)
```

Comment:
I found `into_ref_with_type` is removed here and there. They must be keeped to provide subclass type information.
`.into_ref_with_type(vm, cls).into_pyresult(vm)` is expected

Reply:
actually, i removed all `into_ref_with_type` due to match the signature.

Reply:
but only making the two function to have `into_ref_with_type`, the problem fixed. Can you explain some detail of it?

Reply:
`into_ref_with_type` explicitly set the type of the instance while `into_ref` set the type by its payload. `into_pyresult` internally calls `into_pyobject` which also calls `into_ref` by needs. So `into_ref_with_type` is not called, any subclass of the types will lose its actual type while `into_ref` is implicitly called.

Reply:
i got it thanks! I think removing all `into_ref_with_type` caused so many problems.